### ISSUE TITLE
sys-libs/ncurses-compat-5.9 will not build with lto

### ISSUE THREAD

**Issue by KlausMeier** [InBetweenNames/gentooLTO#418]
CFLAGS="-march=native -O2 ${FLTO} -fuse-linker-plugin -pipe -ftree-vectorize"

checking whether able to link to dl*() functions... configure: error: Cannot link test program for libdl

**Comment by l1am9111** [comment]
Builds with LTO disabled.
`=sys-libs/ncurses-compat-5.9 *FLAGS-=-flto*`

**Comment by KlausMeier** [comment]
Do you think I am stupid?

**Comment by pchome** [comment]
@KlausMeier 
> Do you think I am stupid?

Well, you should provide such information by yourself next time. This issue will be read by people who faced similar problem, ...

Regarding the issue itself, I don't know what `sys-libs/ncurses-compat` for, but obviously it have similar issues as `sys-libs/ncurses`.
[this](https://github.com/InBetweenNames/gentooLTO/blob/master/sys-config/ltoize/files/package.cflags/ltoworkarounds.conf#L116)

**Comment by KlausMeier** [comment]
I report this, because this is a problem. And it should be included in the list of packages, which will not build with lto. I know, that I can build it without lto. But how can this project grow, if nobody reports something? This answer was so stupid.

Why should I report problems? Please, tell me.

**Comment by elsandosgrande** [comment]
@KlausMeiser Well, it would have been a little bit nicer if you had typed the line out yourself, but omitting it was not a big problem in my opinion.

---

@pchome 
1. You should be a little bit nicer to people.
2. ncurses-compat is a replacement for ncurses-5.9 as far as I can tell, since that package got removed when I installed compat.

**Comment by ElDavoo** [comment]
Calm down people. We're all here to make stuff work.
This issue started since Michał Górny decided to make the compat ABI in a separate package (https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=98f0fc212d58518dbe50e962b56a32cf809b0999) and then proceeded to replace ncurses as a dependency with ncurses-compat in many packages, for example android studio (https://gitweb.gentoo.org/repo/gentoo.git/log/?qt=grep&q=ncurses).
In other words, it's a name change, so we just have to copy the same patch that ncurses has. 
The pull request below correctly fixes the issue for me.
